src/main/java/io/vertx/core/Future.java

```diff
@@ -267,6 +267,10 @@ static <T> Future<T> failedFuture(String failureMessage) {
   @Fluent
   Future<T> onComplete(Handler<AsyncResult<T>> handler);
 
+  default Future<T> onComplete(Handler<T> successHandler, Handler<Throwable> failureHandler) {
+      return onSuccess(successHandler).onFailure(failureHandler);
```

Comment:
actually this should be implemented instead as

```java
onComplete(ar -> {
  if (ar.succeeded()) {
  } else {
  }
});
```

as it is more efficient

Reply:
Implemented!